src/plonk/prover.rs

```diff
@@ -53,16 +55,26 @@ pub fn create_proof<
     let mut meta = ConstraintSystem::default();
     let config = ConcreteCircuit::configure(&mut meta);
 
-    struct InstanceSingle<'a, C: CurveAffine> {
-        pub instance_values: &'a [Polynomial<C::Scalar, LagrangeCoeff>],
+    struct InstanceSingle<C: CurveAffine> {
+        pub instance_values: Vec<Polynomial<C::Scalar, LagrangeCoeff>>,
         pub instance_polys: Vec<Polynomial<C::Scalar, Coeff>>,
         pub instance_cosets: Vec<Polynomial<C::Scalar, ExtendedLagrangeCoeff>>,
     }
 
     let instance: Vec<InstanceSingle<C>> = instances
         .iter()
         .map(|instance| -> Result<InstanceSingle<C>, Error> {
-            let instance_commitments_projective: Vec<_> = instance
+            let instance_values = instance
+                .iter()
+                .map(|values| {
+                    let mut poly = domain.empty_lagrange();
+                    for (poly, value) in poly.iter_mut().zip(values.iter()) {
```

Comment:
We should check that `values` is not longer than `poly`, or else this will silently drop the tail of instance columns that are over-long.